commit 4b1b71b187faa18b6011809f9481a949bb54f772
Author: Cole Robinson <crobinso@redhat.com>
Date:   Thu Sep 22 11:44:30 2011 -0400

    virt-install: Fix use of --nonetworks

diff --git a/tests/clitest.py b/tests/clitest.py
index 128e1f2..703e915 100644
--- a/tests/clitest.py
+++ b/tests/clitest.py
@@ -568,6 +568,8 @@ args_dict = {
         "--hvm --pxe --filesystem /foo/source,/bar/target",
         # --filesystem template
         "--hvm --pxe --filesystem template_name,/,type=template",
+        # no networks
+        "--hvm --nodisks --nonetworks --cdrom %(EXISTIMG1)s",
       ],
 
       "invalid": [
@@ -1037,8 +1039,6 @@ args_dict = {
         "--network=user,model=e1000",
         # several networks
         "--network=network:default,model=e1000 --network=user,model=virtio",
-        # no networks
-        #"--nonetworks",
       ],
       "invalid": [
         # Nonexistent network
diff --git a/virt-install b/virt-install
index 1dd97a6..88b44f9 100755
--- a/virt-install
+++ b/virt-install
@@ -390,11 +390,11 @@ def check_option_collisions(options, guest):
 
     # Network collisions
     if options.nonetworks:
-        if options.macs:
+        if options.mac:
             fail(_("Cannot use --mac with --nonetworks"))
-        if options.bridges:
+        if options.bridge:
             fail(_("Cannot use --bridges with --nonetworks"))
-        if options.networks:
+        if options.network:
             fail(_("Cannot use --network with --nonetworks"))
         return
 
