commit 7e4e0f5da18f55fa382260d9939fd415dd0272c9
Author: Cole Robinson <crobinso@redhat.com>
Date:   Thu Sep 22 11:01:42 2011 -0400

    Installer: Never upload media if using local session URI

diff --git a/virtinst/DistroInstaller.py b/virtinst/DistroInstaller.py
index ff7e125..f5ddf5d 100644
--- a/virtinst/DistroInstaller.py
+++ b/virtinst/DistroInstaller.py
@@ -313,9 +313,9 @@ class DistroInstaller(Installer.Installer):
         system_scratchdir = self._get_system_scratchdir()
 
         if (not guest.is_remote() and
-            self.scratchdir == system_scratchdir):
+            (self.is_session_uri() or self.scratchdir == system_scratchdir)):
             # We have access to system scratchdir, don't jump through hoops
-            logging.debug("Have access to local system scratchdir so"
+            logging.debug("Have access to preferred scratchdir so"
                           " nothing to upload")
             return kernel, initrd
 
